src/extensions/replay/sessionrecording.ts

```diff
@@ -464,26 +478,39 @@ export class SessionRecording {
         }
     }
 
-    private _tryRRwebMethod(rrwebMethod: () => void): boolean {
-        if (!this._captureStarted) {
-            return false
-        }
+    private _tryRRWebMethod(queuedRRWebEvent: QueuedRRWebEvent): boolean {
         try {
-            rrwebMethod()
+            queuedRRWebEvent.rrwebMethod()
             return true
         } catch (e) {
             // Sometimes a race can occur where the recorder is not fully started yet
-            logger.error('[Session-Recording] using rrweb when not started.', e)
+            logger.warn('[Session-Recording] could not emit queued rrweb event.', e)
+            this.queuedRRWebEvents.length < 10 &&
+                this.queuedRRWebEvents.push({
+                    enqueuedAt: queuedRRWebEvent.enqueuedAt || Date.now(),
+                    attempt: queuedRRWebEvent.attempt++,
+                    rrwebMethod: queuedRRWebEvent.rrwebMethod,
+                })
             return false
         }
     }
 
     private _tryAddCustomEvent(tag: string, payload: any): boolean {
-        return this._tryRRwebMethod(() => this.rrwebRecord?.addCustomEvent(tag, payload))
+        return this._tryRRWebMethod({
+            // this should throw if rrwebRecord is not available
+            rrwebMethod: () => this.rrwebRecord!.addCustomEvent(tag, payload),
+            enqueuedAt: Date.now(),
+            attempt: 0,
```

Comment:
```suggestion
            attempt: 1,
```

If it throws, it's technically 1 attempt?

Reply:
Same for _tryTakeFullSnapshot